### PR TITLE
Add Dockerfile and scripts to build PAXML on ARM64

### PR DESCRIPTION
**Pull request by joker-eph**

### CODE REVIEW

File: .github/container/install_lingvo_aarch64.sh
Comment (nouiz) [inline .github/container/install_lingvo_aarch64.sh:59]: if -> as?
Do you have some tests failing and so want to skip them for now?

Comment (joker-eph) [reply .github/container/install_lingvo_aarch64.sh:59]: It's been a game of dependency chasing to get all the tests to pass, I got it at some point but it was still fragile, I can give it another try

Comment (nouiz) [reply .github/container/install_lingvo_aarch64.sh:59]: Or just update the comment to retry this when we have a more stable installation instruction?

Comment (joker-eph) [reply .github/container/install_lingvo_aarch64.sh:59]: So test are "mostly" passing already right now, but they are flaky:
- ` //lingvo/core:saver_test` passes 100 times when run alone, but fails when it runs concurrently with other tests...
- `//lingvo/tasks/mt:model_test                                             FAILED in 100 out of 2000 in 57.6s` it's flaky with 5% of the case and the following log on failure:

```
AssertionError: 
Not equal to tolerance rtol=1e-06, atol=0.001
Mismatched value: a is different from b. 
not close where = (array([3]), array([0]))
not close lhs = [181.64984]
not close rhs = [181.65323]
not close dif = [0.00338869]
not close tol = [0.00118165]
dtype = float32, shape = (5, 2)
Mismatched elements: 1 / 10 (10%)
Max absolute difference: 0.00338869
Max relative difference: 1.8654727e-05
 x: array([[226.91527 ,  10.373269],
       [222.4018  ,   9.463906],
       [248.7229  ,   9.899419],...
 y: array([[226.91527 ,  10.373269],
       [222.4018  ,   9.463906],
       [248.72293 ,   9.89942 ],...
```

Comment (joker-eph) [reply .github/container/install_lingvo_aarch64.sh:59]: I just excluded these two tests explicitly by patching the `bazel test` invocation, that we we'll catch regression for other tests.

Comment (nouiz) [reply .github/container/install_lingvo_aarch64.sh:59]: Great. They seems to be tests issue, not issue with ARM. So disabling them is good.

Comment (nouiz) [reply .github/container/install_lingvo_aarch64.sh:59]: I don't see the code that disable those tests. Where it is?

**Review by nouiz — APPROVED** [review]
LGTM, only questions that shouldn't block the merge.

File: .github/container/Dockerfile.aarch64.pax
Comment (joker-eph) [reply .github/container/Dockerfile.aarch64.pax]: Cargo is the Rust packaging system.

`pip install t5` builds the wheel on the fly, and Rust is a dep here.

File: .github/container/install_lingvo_aarch64.sh
Comment (joker-eph) [reply .github/container/install_lingvo_aarch64.sh]: I didn't amend before pushing... PTAL

**Review by nouiz — APPROVED** [review]
LGTM.

**Review by yhtang — CHANGES_REQUESTED** [review]
Given that this PR contains many temporary fixes that will eventually go away if the related PRs are properly upstreamed, can we create a standalone `install-pax-aarch64.sh` script, and leave the original `intall-pax.sh` as-is?

File: .github/container/install-pax.sh
Comment (joker-eph) [inline .github/container/install-pax.sh:112]: This will likely stay "forever"

File: .github/container/install-pax.sh
Comment (joker-eph) [inline .github/container/install-pax.sh:131]: This will likely stay "forever" as well

File: .github/container/install-pax.sh
Comment (joker-eph) [inline .github/container/install-pax.sh:152]: This will likely stay "forever": Google does not intend to provide an arm64 wheel here.

I could pull this into another script if you'd like though!

**Comment by joker-eph** [comment]
> Given that this PR contains many temporary fixes that will eventually go away if the related PRs are properly upstreamed, 

The PRs are not referenced in install-pax but in a new script install-lingvo.sh, can you take another look to be sure you didn't mix up the two files here?


> can we create a standalone install-pax-aarch64.sh script, and leave the original intall-pax.sh as-is?

Happy to fork the script, that's what I did originally, and I ended up heavily refactoring to minimize the change because I assume we wouldn't want to maintain two scripts!
I commented in the install-pax.sh on the changes: I think they will all stay (but it's just some "sed" pre-processing before install). The diff could be reduced by moving the installation of array-record into its own script though.

**Comment by yhtang** [comment]
Thanks for the clarification! Let's go ahead and merge the PR.

As a side topic, do we have a plan or any ongoing discussion regarding a more systematic approach to build/host those arm64 wheels that their original maintainers do not intend to maintain?

**Review by yhtang — APPROVED** [review]

**Comment by joker-eph** [comment]
> Thanks for the clarification! Let's go ahead and merge the PR.

OK, I'll merge as-is then.

> As a side topic, do we have a plan or any ongoing discussion regarding a more systematic approach to build/host those arm64 wheels that their original maintainers do not intend to maintain?

We haven't talked about it, we could collaborate with Google to setup an automatic builder which could push nightly wheels for example. Or just ask them to coordinate for every new release. I'm not sure what's the best option?

**Pull request merged**
